src/screens/WalletDiagnosticsSheet.js

```diff
@@ -0,0 +1,327 @@
+        <Bold>Secret:</Bold> {` `}
+      </Text>
+      <RowWithMargins>
+        <SecretInput color={colors.blueGreyDark} value={data.secret} />
+      </RowWithMargins>
+      <ButtonPressAnimation onPress={handlePressRestore}>
+        <View
+          backgroundColor={colors.dpiMid}
+          borderRadius={15}
+          style={{ paddingHorizontal: 15, paddingVertical: 10 }}
+        >
+          <Text align="center" weight="bold">
+            Restore
+          </Text>
+        </View>
+      </ButtonPressAnimation>
+      <Divider />
+    </ColumnWithMargins>
+  );
+};
+
+const WalletDiagnosticsSheet = () => {
+  const { height: deviceHeight, width: deviceWidth } = useDimensions();
+  const { colors } = useTheme();
+  const { goBack } = useNavigation();
+  const [keys, setKeys] = useState();
+
+  const walletsWithBalancesAndNames = useWalletsWithBalancesAndNames();
+
+  useEffect(() => {
+    const init = async () => {
+      try {
+        const allKeys = await loadAllKeys();
+        const processedKeys = await Promise.all(
+          allKeys
+            .filter(
+              key =>
+                key?.username?.indexOf(seedPhraseKey) !== -1 ||
+                key?.username?.indexOf(privateKeyKey) !== -1
+            )
+            .map(async key => {
+              const secretObj = JSON.parse(key.password);
+              const secret = secretObj.seedphrase || secretObj.privateKey;
+              const {
+                address,
+                type,
+              } = await ethereumUtils.deriveAccountFromWalletInput(secret);
+              let createdAt = null;
+              let label = null;
+              Object.keys(walletsWithBalancesAndNames).some(k => {
+                const found = walletsWithBalancesAndNames[k].addresses.some(
+                  account => {
+                    if (toLower(account.address) === toLower(address)) {
+                      label = account.label || account.ens;
+                      return true;
+                    }
+                    return false;
+                  }
+                );
+                return found;
+              });
+
+              if (key?.username?.indexOf(`_${seedPhraseKey}`) !== -1) {
+                const tsString = key.username
+                  .replace('wallet_', '')
+                  .replace(`_${seedPhraseKey}`, '');
+                const ts = new Date(Number(tsString));
+                createdAt = ts.toString();
+              }
+
+              return {
+                ...key,
+                address,
+                createdAt,
+                label,
+                secret,
+                type,
+              };
+            })
+        );
+        setKeys(processedKeys);
+      } catch (e) {
+        logger.sentry('Error processing keys for wallet diagnostics', e);
+        const customError = new Error('WalletDiagnostics init failed');
+        captureException(customError);
+      }
+    };
+    setTimeout(() => {
```

Comment:
@brunobar79 do you mind adding a comment before the setTimeout that we're doing this bc the flow is coming from a user tapping AppVersionStamp a bunch of times and we don't want them accidentally restoring

Reply:
wondering if we can just remove the tapping aspect? its kinda finicky already and i doubt anyone taps that on accident

Reply:
nah, it works fine now